### PR TITLE
bugfix(authentication): fix config errors on initialization

### PR DESCRIPTION
**Pull request by ntruchsess**
corresponding CD-PR: https://github.com/catenax-ng/tx-portal-cd/pull/77
* add healthchecks to ensure pod is finished initializing config before facing traffic
Refs: CPLP-2567

### CODE REVIEW

File: src/framework/Framework.Web/JwtBearerConfigurationHealthCheck.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private readonly IConfigurationManager<OpenIdConnectConfiguration> _configurationManager;
+
+    public JwtBearerConfigurationHealthCheck(IOptions<JwtBearerOptions> jwtOptions)
+    {
+        var options = jwtOptions.Value;
+        if (!options.RequireHttpsMetadata)
+        {
+            options.BackchannelHttpHandler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (_,_,_,_) => true
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/framework/Framework.Web/JwtBearerConfigurationHealthCheck.cs:40]: this is a false-positive as sonar ignores the condition that ensures this code is not running on any non-test-environment.

File: src/framework/Framework.Web/StartupServiceWebApplicationExtensions.cs [hunk 2]
```diff
@@ -73,6 +73,7 @@ public static WebApplication CreateApp<TProgram>(this WebApplication app, string
         app.UseAuthorization();
 
         app.MapControllers();
+        app.MapDefaultHealthChecks(app.Configuration.GetSection("HealthChecks").Get<IEnumerable<HealthCheckSettings>>());
```
Comment (Phil91) [inline src/framework/Framework.Web/StartupServiceWebApplicationExtensions.cs:76]: we should probably check that the same path isn't used twice

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/framework/Framework.Web/StartupServiceWebApplicationExtensions.cs (hunk 2 above)
Comment (ntruchsess) [reply src/framework/Framework.Web/StartupServiceWebApplicationExtensions.cs:76]: done

**Comment by ntruchsess** [comment]
Unfortunately WebApplication is a sealed class that cannot be faked, therefore unit-test-coverage cannot easily be increased.

**Review by Phil91 — APPROVED** [review]

**Pull request merged**